classes/nameable.rb

```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
```

Comment:
- [ OPTIONAL ] It's truly impressive that you've devised a method for triggering a "NotImplementedError." However, your current implementation lacks a crucial element: error messages, along with information about the class and method responsible for raising the error. 🚫🚧📢 It would be immensely beneficial to users if this essential information were included when an error occurs, so they can easily pinpoint the issue and its source. 🕵️‍♂️💡

```suggestion
 def correct_name
        raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
    end
```